### PR TITLE
Set external signer for requests payload

### LINKED ISSUE #373: Provide support for external request signing (extract private key from client)
Currently, it seems in order to make any request to NeoFS (using the SDK), you have to pass private key data to the client to sign messages. However, I'd like to keep away my private key data from this code and create the necessary signature elsewhere, e.g., in a cold wallet.

For example, to create a new container, I image a process like the following:

1. Prepare a request to create a container and get the (stable marshalled) request data of it.
2. Sign the request data in a separate environment.
3. Build the request based on the request data and the signature data and then send it within the SDK.

Can you tell me if this is somehow possible to do already and I missed it? Otherwise, I think this would be a very useful feature.

@gsmachado @cthulhu-rider

### PR DESCRIPTION
**Pull request by smallhive**
close #373

### CODE REVIEW

File: client/container.go [hunk 1]
```diff
@@ -430,23 +459,25 @@ func (c *Client) ContainerDelete(ctx context.Context, prm PrmContainerDelete) (*
...
 
-	err := sig.Calculate(neofsecdsa.SignerRFC6979(c.prm.key), data)
-	if err != nil {
+	var sig neofscrypto.Signature
+	if err := sig.Calculate(signer, data); err != nil {
 		return nil, fmt.Errorf("calculate signature: %w", err)
 	}
 
 	var sigv2 refs.Signature
-
```
Comment (roman-khimov) [inline client/container.go:449]: Avoid doing whitespace changes that are not related to the current problem.

File: client/container.go [hunk 2]
```diff
@@ -91,20 +105,25 @@ func (c *Client) ContainerPut(ctx context.Context, prm PrmContainerPut) (*ResCon
 		panic(panicMsgMissingContainer)
 	}
 
-	// TODO: check private key is set before forming the request
-	// sign container
-	var cnr v2container.Container
+	var (
+		cnr    v2container.Container
+		signer neofscrypto.Signer
```
Comment (roman-khimov) [inline client/container.go:110]: `signer = prm.signer` +

File: client/container.go (hunk 1 above)
Comment (roman-khimov) [reply client/container.go:449]: Or do them in a separate commit that doesn't change any logic.

File: client/container.go [hunk 3]
```diff
@@ -375,6 +394,9 @@ type PrmContainerDelete struct {
 
 	tokSet bool
 	tok    session.Container
+
+	signerSet bool
```
[the remark this answers is not in the document]
Comment (smallhive) [reply client/container.go]: `signerSet` removed
